richtextfx/src/main/java/org/fxmisc/richtext/model/CustomObject.java

```diff
@@ -0,0 +1,82 @@
+package org.fxmisc.richtext.model;
+
+
+/**
+ * This is the base class for custom objects in the model layer.
+ * Its String representation is always one character long and contains
+ * the "object replacement character" (\ufffc). 
+ */
+public abstract class CustomObject<S> implements Segment<S> {
+
+    private S style;
+    private SegmentType typeId;
```

Comment:
A minor detail, but this should probably be renamed to `type` now, not `typeId` as it was when you were using an `int` rather than your current enum/interface approach